File: src/main/java/model/types/Course.java
package model.types;
import java.util.ArrayList;

import model.database.DBData;
import model.database.DBProvider;
import model.database.DBUniqueID;
import model.database.IDTypeMismatchExcception;
import model.database.InitException;

public class Course extends DBData {
	public ArrayList<Course> prereqs = new ArrayList<Course>();
	public ArrayList<Course> antireqs = new ArrayList<Course>();
	
	private int courseNumber;
	private String departmentCode;
	private DBUniqueID departmentID;
	
	public Course(DBUniqueID id) throws IDTypeMismatchExcception {
		super(id,"Course");
	}
	
	public Department Department() throws DataNotFoundException, IDTypeMismatchExcception {
		try {
			return DBProvider.Instance().GetConnection().GetDepFromCode(departmentID);
		} catch (InitException e) {
			throw new DataNotFoundException();
		}
	}
	public String Title() {
		return departmentCode+" "+courseNumber;
	}
}

File: src/main/kotlin/model/types/Course.kt
package model.types

import model.database.*
import java.util.*

class Course @Throws(IDTypeMismatchExcception::class)
constructor(id: DBUniqueID) : DBData(id, EDBTypeCode.COURSE)
{
    var departmentID : DBUniqueID = DBUniqueID(EDBTypeCode.DEPARTMENT)
    var code : String = ""
    var number : Int = 0

    var year : Int = 2000

    private var prereqs = ArrayList<DBUniqueID>()
    private var antireqs = ArrayList<DBUniqueID>()

    fun addPreReq(id : DBUniqueID){
        if (id.checkType(EDBTypeCode.COURSE) && !(id in prereqs)) {
            prereqs.add(id)
        }
    }
    fun addAntiReq(id : DBUniqueID){
        if (id.checkType(EDBTypeCode.COURSE) && !(id in antireqs)) {
            antireqs.add(id)
        }
    }
    fun getAntiReqs() = Collections.unmodifiableList(antireqs)

    fun getPreReqs() = Collections.unmodifiableList(prereqs)

    fun removeAntiReq(id : DBUniqueID) = antireqs.remove(id)

    fun removePreReq(id : DBUniqueID) = prereqs.remove(id)

    fun getName(): String {
        return code
    }
    var title = ""
    var description = ""
}
